mailbox/tools/indexer/src/main/java/org/apache/mailbox/tools/indexer/RunningOptionsDTO.java

```diff
@@ -28,22 +28,33 @@
 
 public class RunningOptionsDTO {
     public static RunningOptionsDTO toDTO(RunningOptions runningOptions) {
-        return new RunningOptionsDTO(Optional.of(runningOptions.getMessagesPerSecond()));
+        return new RunningOptionsDTO(
+            Optional.of(runningOptions.getMessagesPerSecond()),
+            Optional.of(runningOptions.getMode()));
     }
 
     private final Optional<Integer> messagesPerSecond;
+    private final Optional<RunningOptions.Mode> mode;
 
     @JsonCreator
-    public RunningOptionsDTO(@JsonProperty("messagesPerSecond") Optional<Integer> messagesPerSecond) {
+    public RunningOptionsDTO(@JsonProperty("messagesPerSecond") Optional<Integer> messagesPerSecond,
+                             @JsonProperty("mode") Optional<RunningOptions.Mode> mode) {
         this.messagesPerSecond = messagesPerSecond;
+        this.mode = mode;
     }
 
     public Optional<Integer> getMessagesPerSecond() {
         return messagesPerSecond;
     }
 
+    public Optional<RunningOptions.Mode> getMode() {
+        return mode;
+    }
+
     public RunningOptions toDomainObject() {
-        return new RunningOptions(messagesPerSecond.orElse(RunningOptions.DEFAULT.getMessagesPerSecond()));
+        return new RunningOptions(
+            messagesPerSecond.orElse(RunningOptions.DEFAULT.getMessagesPerSecond()),
+            mode.orElse(RunningOptions.DEFAULT.getMode()));
```

Comment:
I think we need a separated DEFAULT_MODE constant that RunningOptions.DEFAULT would then depend on.